dcspy/models.py

```diff
@@ -899,16 +903,20 @@ def bytes_requests(self, key_down: Optional[int] = None) -> List[bytes]:
         :param key_down: 1 indicate when G-Key was push down, 0 when G-Key is up
         :return: a list of bytes representing the individual requests
         """
-        if self.is_cycle:
+        if self.is_push_button and isinstance(self.key, Gkey):
+            request = f'{self.ctrl_name} {key_down}\n'
+        elif self.is_push_button and isinstance(self.key, LcdButton):
+            request = f'{self.ctrl_name} {KEY_DOWN}\n|{self.ctrl_name} {KEY_UP}\n'
+        # Ignore key up events for non-push button requests so that
+        # these are not triggered again when the G-Key is released.
+        elif key_down is None or key_down == KEY_UP:
```

Comment:
This was the bug. When you release the G-key it would come through this block, even though key_down was 0, and cause the request to be triggered again. This would mean buttons would be cycled twice etc.